pkg/registry/seedmanagement/managedseedset/strategy.go

```diff
@@ -0,0 +1,157 @@
+	return false
+}
+
+// AllowUnconditionalUpdate returns true if the object can be updated
+// unconditionally (irrespective of the latest resource version), when
+// there is no resource version specified in the object.
+func (Strategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+// ValidateUpdate validates the update on the given old and new object.
+func (Strategy) ValidateUpdate(ctx context.Context, newObj, oldObj runtime.Object) field.ErrorList {
+	oldManagedSeedSet, newManagedSeedSet := oldObj.(*seedmanagement.ManagedSeedSet), newObj.(*seedmanagement.ManagedSeedSet)
+	return validation.ValidateManagedSeedSetUpdate(newManagedSeedSet, oldManagedSeedSet)
+}
+
+// StatusStrategy defines the strategy for storing seeds statuses.
+type StatusStrategy struct {
+	Strategy
+}
+
+// NewStatusStrategy defines the storage strategy for the status subresource of ManagedSeedSets.
+func NewStatusStrategy() StatusStrategy {
+	return StatusStrategy{NewStrategy()}
+}
+
+// PrepareForUpdate is invoked on update before validation to normalize
+// the object.  For example: remove fields that are not to be persisted,
+// sort order-insensitive list fields, etc.  This should not remove fields
+// whose presence would be considered a validation error.
+func (s StatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newManagedSeedSet := obj.(*seedmanagement.ManagedSeedSet)
+	oldManagedSeedSet := old.(*seedmanagement.ManagedSeedSet)
+	newManagedSeedSet.Spec = oldManagedSeedSet.Spec
+}
+
+// ValidateUpdate validates the update on the given old and new object.
+func (StatusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
+	return validation.ValidateManagedSeedSetStatusUpdate(obj.(*seedmanagement.ManagedSeedSet), old.(*seedmanagement.ManagedSeedSet))
+}
+
+// MatchManagedSeedSet returns a generic matcher for a given label and field selector.
+func MatchManagedSeedSet(label labels.Selector, field fields.Selector) storage.SelectionPredicate {
+	return storage.SelectionPredicate{
+		Label:    label,
+		Field:    field,
+		GetAttrs: GetAttrs,
+	}
+}
+
+// GetAttrs returns labels and fields of a given object for filtering purposes.
+func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
+	managedSeedSet, ok := obj.(*seedmanagement.ManagedSeedSet)
+	if !ok {
+		return nil, nil, fmt.Errorf("not a ManagedSeedSet")
+	}
+	return labels.Set(managedSeedSet.ObjectMeta.Labels), ToSelectableFields(managedSeedSet), nil
+}
+
+// ToSelectableFields returns a field set that represents the object.
+func ToSelectableFields(managedSeedSet *seedmanagement.ManagedSeedSet) fields.Set {
+	// The purpose of allocation with a given number of elements is to reduce
+	// amount of allocations needed to create the fields.Set. If you add any
+	// field here or the number of object-meta related fields changes, this should
+	// be adjusted.
+	shootSpecificFieldsSet := make(fields.Set, 3)
```

Comment:
We don't have any fields in addition to the selectable metadata fields (`namespace`, `name`).
Apart from that, you can maybe rename this var.
```suggestion
	shootSpecificFieldsSet := make(fields.Set, 2)
```

Reply:
OK, thanks.